app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -24,13 +24,26 @@
     var canI = $filter('canI');
     var humanizeKind = $filter('humanizeKind');
     var uniqueId = _.uniqueId();
+    var keyValidator = new RegExp("^[A-Za-z_][A-Za-z0-9_]*$");
```

Comment:
Better to use a regex literal. It improves performance, so it's not just style. (I realize this was probably copied from elsewhere.)

```js
var keyValidator = /^[A-Za-z_][A-Za-z0-9_]*$/;
```